src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,122 @@
-import React from 'react';
+import React, { Component } from 'react';
+import cn from 'classnames';
+import PropTypes from 'prop-types';
+
 import './TodoList.scss';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends Component {
+  state = {
+    postId: 0,
+    filterStatus: 'all',
+    filterTitle: '',
+  };
+
+  buttonClickHandler = (postId, userId) => {
+    this.setState({ postId });
+
+    this.props.onSelectUser(userId);
+  }
+
+  setFilter = (event) => {
+    this.setState({ filterStatus: event.target.value });
+  }
+
+  onInputHandler = (event) => {
+    this.setState({ filterTitle: event.target.value });
+  }
+
+  filterByStatus = (todo) => {
+    const { filterStatus } = this.state;
+
+    switch (filterStatus) {
+      case 'completed': return todo.completed;
+      case 'active': return !todo.completed;
+      default: return true;
+    }
+  }
+
+  // eslint-disable-next-line consistent-return
+  filterByTitle = (todo) => {
+    const { filterTitle } = this.state;
+
+    if (todo.title !== null) {
+      return todo.title.toLowerCase().includes(filterTitle.toLowerCase());
+    }
+  }
+
+  render() {
+    const { todos } = this.props;
+    const { postId } = this.state;
+
+    const filteredTodos = todos
+      .filter(this.filterByStatus)
+      .filter(this.filterByTitle);
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <div className="TodoList__list-container">
+          <form>
+            <label className="TodoList__label">
+              Todo filter:
+              <input
+                onChange={event => this.onInputHandler(event)}
```

Comment:
if you need to pass just event, no need to use arrow func, just pass callback

```suggestion
                onChange={this.onInputHandler}
```